### PR TITLE
cpu: add loongarch64 support

### PR DESCRIPTION
**Pull request by loong-hy**
about loongarch：https://docs.kernel.org/arch/loongarch/introduction.html

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,10 +74,10 @@ oneDNN supports platforms based on the following architectures:
 - [OpenPOWER](https://openpowerfoundation.org/) / [IBM Power ISA](https://en.wikipedia.org/wiki/Power_ISA).
 - [IBMz z/Architecture (s390x)](https://en.wikipedia.org/wiki/Z/Architecture).
 - [RISC-V 64-bit (RV64)](https://en.wikipedia.org/wiki/RISC-V).
-
+- [LOONGARCH 64 bit](https://docs.kernel.org/arch/loongarch/introduction.html).
```
Comment (dzarukin) [inline README.md:77]: Please restore the new line to separate the warning.

File: src/cpu/platform.hpp [hunk 2]
```diff
@@ -30,11 +30,12 @@
 // - DNNL_PPC64
 // - DNNL_S390X
 // - DNNL_RV64
+// - DNNL_LOONGARCH64
 // - DNNL_ARCH_GENERIC
 // Target architecture macro is set to 1, others to 0. All macros are defined.
 
 #if defined(DNNL_X64) + defined(DNNL_AARCH64) + defined(DNNL_PPC64) \
-                + defined(DNNL_S390X) + defined(DNNL_RV64) \
+                + defined(DNNL_S390X) + defined(DNNL_RV64) + defined(DNNL_LOONGARCH64) \
```
Comment (dzarukin) [inline src/cpu/platform.hpp:38]: Please apply clang-format on this file.

Comment (loong-hy) [reply src/cpu/platform.hpp]: > Please apply clang-format on this file.

It's been revised and resubmitted, thanks

File: cmake/platform.cmake [hunk 3]
```diff
@@ -363,6 +363,8 @@ elseif(UNIX OR MINGW)
         elseif(DNNL_TARGET_ARCH STREQUAL "RV64")
             # G = General-purpose extensions, C = Compression extension (very common).
             append(DEF_ARCH_OPT_FLAGS "-march=rv64gc")
+	elseif(DNNL_TARGET_ARCH STREQUAL "LOONGARCH64")
+            append(DEF_ARCH_OPT_FLAGS "-march=loongarch64")
```
Comment (mgouicem) [inline cmake/platform.cmake:367]: Any reason to not add opt flags for compilers other than gcc ?

**Review by mgouicem — COMMENTED** [review]
thanks for the PR. I have few questions:
- could you reword the commit message to align with oneDNN guidelines (see [here](https://github.com/oneapi-src/oneDNN/blob/main/CONTRIBUTING.md#code-contribution-guidelines). Something like `cpu: add loongarch64 support` should do.
- In general, adding a new CPU to oneDNN platform.cmake is necessary only when isa specific optimization or fixes are added to the library. From the look of the patch, it does not seem to be the case for loongarch64 (this patch is equivalent to building the library with `ONEDNN_ARCH_OPT_FLAGS=-march=loongarch64` cmake option). Do you have any plan to add loongarch64 isa specific code to oneDNN?

**Comment by loong-hy** [comment]
> thanks for the PR. I have few questions:
> 
> * could you reword the commit message to align with oneDNN guidelines (see [here](https://github.com/oneapi-src/oneDNN/blob/main/CONTRIBUTING.md#code-contribution-guidelines). Something like `cpu: add loongarch64 support` should do.

Hello, I have changed the commit as requested.

> * In general, adding a new CPU to oneDNN platform.cmake is necessary only when isa specific optimization or fixes are added to the library. From the look of the patch, it does not seem to be the case for loongarch64 (this patch is equivalent to building the library with `ONEDNN_ARCH_OPT_FLAGS=-march=loongarch64` cmake option). Do you have any plan to add loongarch64 isa specific code to oneDNN?

Because I will add loongarch related code later, I didn't change it this time because of time constraints.

**Comment by mgouicem** [comment]
@loong-hy do you mind submitting this patch as part of the future PR that will add loongarch specific code?
This will allow to introduce these knobs only when actually necessary. Thanks.

**Comment by loong-hy** [comment]
> @loong-hy do you mind submitting this patch as part of the future PR that will add loongarch specific code? This will allow to introduce these knobs only when actually necessary. Thanks.

Sure, I'll revise and resubmit the code shortly, thanks!

**Review by ranukund — APPROVED** [review]
Looks good, thanks!

**Comment by vpirogov** [comment]
Closing as stale.

**Pull request closed without merging**